<?xml version="1.0"?>
<root main_tree_to_execute="ExplorationTree">
    <!-- ////////// -->
    <BehaviorTree ID="ExplorationTree">
        <RetryUntilSuccessful num_attempts="-1">
            <ForceFailure>
                <Sequence>
                    <Fallback name="Exploration Request">
                        <Action ID="GetExplorationRequest"/>
                        <Condition ID="IsExplorationRequired"/>
                    </Fallback>
                    <Sequence name="LocomotionToTarget">
                        <Fallback name="DefineNavTarget">
                            <Action ID="CollectObjectPose"/>
                            <Sequence name="Explore">
                                <Action ID="CollectFrontiers"/>
                                <Action ID="Explore"/>
                            </Sequence>
                        </Fallback>
                        <Fallback name="DriveToTarget">
                            <Action ID="CheckLocomotionStatus"/>
                            <Action ID="SendNavPose"/>
                        </Fallback>
                    </Sequence>
                </Sequence>
            </ForceFailure>
        </RetryUntilSuccessful>
    </BehaviorTree>
    <!-- ////////// -->
    <TreeNodesModel>
        <Action ID="CheckLocomotionStatus"/>
        <Action ID="CollectFrontiers"/>
        <Action ID="CollectObjectPose"/>
        <Action ID="Explore"/>
        <Action ID="GetExplorationRequest"/>
        <Condition ID="IsExplorationRequired"/>
        <Action ID="SendNavPose"/>
    </TreeNodesModel>
    <!-- ////////// -->
</root>
